timestring/Range.py

```diff
@@ -33,25 +33,23 @@ def __init__(self, start, end=None, offset=None, start_of_week=0, tz=None, verbo
             end = str(end)
 
         if start and end:
-            """start and end provided
-            """
+            # Start and end provided as Dates
             self._dates = (Date(start, tz=tz), Date(end, tz=tz))
 
         elif start == 'infinity':
-            # end was not provided
+            # End was not provided
             self._dates = (Date('infinity'), Date('infinity'))
 
         elif re.search(r'(\s(and|to)\s)', start):
-            """Both sides where provided in the start
-            """
+            # Both sides aer provided in string "start"
             start = re.sub('^(between|from)\s', '', start.lower())
             # Both arguments found in start variable
             r = tuple(re.split(r'(\s(and|to)\s)', start.strip()))
-            self._dates = (Date(r[0], tz=tz), Date(r[-1], tz=tz))
+            start = Range(r[0], tz=tz).start
```

Comment:
what is the use-case for this change?

Reply:
Month names and week days parsed by Date do not start on midnight, while we want them to start on midnight in a date range. Maybe we should change `Date`.